src/Pagination.js

```diff
@@ -15,7 +17,11 @@ const Pagination = React.createClass({
     last: React.PropTypes.bool,
     prev: React.PropTypes.bool,
     next: React.PropTypes.bool,
-    onSelect: React.PropTypes.func
+    onSelect: React.PropTypes.func,
+    /**
+     * You can use a custom element for this component
+     */
+    componentClass: CustomPropTypes.elementType
```

Comment:
since this is not being used as a custom element for _this_ component but is being passed down I would name it something a bit more descriptive like `buttonComponentClass`